### DISCUSSION TITLE
Why cant you subscribe to read model updates?

### DISCUSSION THREAD

**Discussion by mikecann** [reimagined/resolve#1774] [General]
As far as I understand it from the docs its only possible to subscribe to view model updates and not read models. Is there a reason for this and is there any plans to allow subscriptions to read models too?

**Comment by superroma** [comment]
View Model is a special kind of read model, which does not have any storage or db, it is built from the event stream on the fly by reducer function.
It is not practical to reduce a stream of all app's events, so View Model by definition is bound to one or several aggregate IDs.
So having aggregateIDs, list of required event types, and a pure reducer function allows resolve to subscribe to those events on the client and apply them (using reducer function) to keep view model up-to date without re-quierying.

Generic read model has some storage, listen to events from all aggregates, and its projection functions are not reducers, these are non-pure functions that mutate the storage. So it is not practical to send applied events to the client, there is no easy way to understand how read model were affected, and there are security issues linked to such a solution.

So there is no generic way to keep result of query to read model in sync with changes being done in that read model.

We thought about notifications from read models. This feature is in plans, but not implemented yet.

**Reply by mikecann** [reply]
I suppose if you use firebase as your DB for your read model you could just then use firebase's native subscribe functionality. Or use DynamoDB and AppSync for something simmilar on AWS.

**Reply by superroma** [reply]
Exactly, in all such cases, if you need something reSolve cannot do - you can talk to particular storage system natively and use its unique features. 

Alternatively, if you know what you are doing, you may do optimistic UI update: if command succeded, it returns a generated event, so you can use it immediately to update the UI

**Reply by mikecann** [reply]
> Alternatively, if you know what you are doing, you may do optimistic UI update: if command succeded, it returns a generated event, so you can use it immediately to update the UI

True, proably best to use both. 

Okay thanks.
